src/nat/middleware/defense_middleware_pii.py

```
@@ -0,0 +1,325 @@
+            try:
+                from presidio_analyzer import AnalyzerEngine
+                from presidio_anonymizer import AnonymizerEngine
+
+                self._analyzer = AnalyzerEngine()
+                self._anonymizer = AnonymizerEngine()
+                logger.info("Presidio engines loaded successfully")
+            except ImportError:
+                raise ImportError(
+                    "Microsoft Presidio is not installed. "
+                    "Install it with: pip install presidio-analyzer presidio-anonymizer"
+                )
```

Ruff (0.14.8)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)